### PR TITLE
Added PhantomJS script : verified on Ubuntu 16.04, RHEL 7.3

### PR DESCRIPTION
**Pull request by sid226**
Logs for UBUNTU 16.04  [phantomjs-2.1.1-2018-10-11-11%3A52%3A08.log](https://github.com/imdurgadas/scripts/files/2469003/phantomjs-2.1.1-2018-10-11-11.3A52.3A08.log)
Logs for RHEL 7.3  [phantomjs-2.1.1-2018-10-11-12%3A26%3A10.log](https://github.com/imdurgadas/scripts/files/2469007/phantomjs-2.1.1-2018-10-11-12.3A26.3A10.log)

### CODE REVIEW

**Comment by sid226** [comment]
Logs For UBUNTU 18.04 [phantomjs-2.1.1-2018-10-11-14%3A00%3A44.log](https://github.com/imdurgadas/scripts/files/2469338/phantomjs-2.1.1-2018-10-11-14.3A00.3A44.log)
Logs for RHEL 7.4 [phantomjs-2.1.1-2018-10-11-13%3A53%3A21.log](https://github.com/imdurgadas/scripts/files/2469341/phantomjs-2.1.1-2018-10-11-13.3A53.3A21.log)

File: PhantomJS/install.sh [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+# © Copyright IBM Corporation 2017, 2018.
+# LICENSE: Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+
+set -e
+
+PACKAGE_NAME="phantomjs"
+PACKAGE_VERSION="2.1.1"
+CURDIR="$(pwd)"
+LOG_FILE="${CURDIR}/${PACKAGE_NAME}-${PACKAGE_VERSION}-$(date +"%F-%T").log"
+OVERRIDE=false
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: Replace `OVERRIDE` with `FORCE`

File: PhantomJS/install.sh [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	rm -rf "${BUILD_DIR}/curl/mk-ca-bundle.pl"
+	rm -rf "${BUILD_DIR}/phantomjs"
+	printf -- 'Cleaned up the artifacts\n' >>"$LOG_FILE"
+
+}
+
+function configureAndInstall() {
+	printf -- 'Configuration and Installation started \n'
+
+	if [[ "${OVERRIDE}" == "true" ]]; then
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: Why override ? We removed this right ?

File: PhantomJS/install.sh [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+		printf -- 'phantomjs exists on the system. Override flag is set to true hence updating the same\n ' | tee -a "$LOG_FILE"
+	fi
+
+	if [[ "${VERSION_ID}" == "15" ]]; then
+		# Build OpenSSL 1.0.2
+		cd "$BUILD_DIR"
+		git clone git://github.com/openssl/openssl.git
+		cd openssl
+		git checkout OpenSSL_1_0_2l
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: combine git clone adn checkout into single command
`git clone -q -b OpenSSL_1_0_2l  git://github.com/openssl/openssl.git`

File: PhantomJS/install.sh [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+		./config --prefix=/usr --openssldir=/usr/local/openssl shared
+		make
+		sudo make install
+
+		# Build cURL 7.52.1
+		cd "$BUILD_DIR"
+		git clone git://github.com/curl/curl.git
+		cd curl
+		git checkout curl-7_52_1
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: same comment as above for clone and checkout. also use -q for silent

File: PhantomJS/install.sh [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+		./buildconf
+		./configure --prefix=/usr/local --with-ssl --disable-shared
+		make && sudo make install
+		export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib64
+		export PATH=/usr/local/bin:$PATH
+		printf -- 'Build cURL success\n' >>"$LOG_FILE"
+
+		# Generate ca-bundle.crt for curl
+		echo insecure >>$HOME/.curlrc
+		wget https://raw.githubusercontent.com/curl/curl/curl-7_53_0/lib/mk-ca-bundle.pl
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: use -q

File: PhantomJS/install.sh [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+
+		printf -- 'Build OpenSSL success\n' >>"$LOG_FILE"
+
+	fi
+
+	# Install Phantomjs
+	cd "$BUILD_DIR"
+	git clone git://github.com/ariya/phantomjs.git
+	cd phantomjs
+	git checkout 2.1.1
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: Same as above. combine clone , checkout with -q

File: PhantomJS/install.sh [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+checkPrequisites #Check Prequisites
+
+DISTRO="$ID-$VERSION_ID"
+case "$DISTRO" in
+"ubuntu-16.04" | "ubuntu-18.04")
+	printf -- "Installing %s %s for %s \n" "$PACKAGE_NAME" "$PACKAGE_VERSION" "$DISTRO" | tee -a "$LOG_FILE"
+	sudo apt-get update >/dev/null
+
+	printf -- 'Installing the PhantomJS from repository \n' | tee -a "$LOG_FILE"
+	sudo sudo apt-get install -y phantomjs >/dev/null
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: add -qq

File: PhantomJS/install.sh [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+	verify_repo_install
+	;;
+
+"rhel-7.3" | "rhel-7.4" | "rhel-7.5")
+	printf -- "Installing %s %s for %s \n" "$PACKAGE_NAME" "$PACKAGE_VERSION" "$DISTRO" | tee -a "$LOG_FILE"
+	printf -- 'Installing the dependencies for PhantomJS from repository \n' | tee -a "$LOG_FILE"
+	sudo yum -y install gcc gcc-c++ make flex bison gperf ruby openssl-devel freetype-devel fontconfig-devel libicu-devel sqlite-devel libpng-devel libjpeg-devel libXfont.s390x libXfont-devel.s390x xorg-x11-utils.s390x xorg-x11-font-utils.s390x tzdata.noarch tzdata-java.noarch xorg-x11-fonts-Type1.noarch xorg-x11-font-utils.s390x python python-setuptools git wget tar >/dev/null
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: add -q

File: PhantomJS/install.sh [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+	configureAndInstall
+	;;
+
+"sles-12.3" | "sles-15")
+	printf -- "Installing %s %s for %s \n" "$PACKAGE_NAME" "$PACKAGE_VERSION" "$DISTRO" | tee -a "$LOG_FILE"
+	printf -- 'Installing the dependencies for PhantomJS from repository \n' | tee -a "$LOG_FILE"
+
+	if [[ "${VERSION_ID}" == "12.3" ]]; then
+		sudo zypper install -y gcc gcc-c++ make flex bison gperf ruby openssl-devel freetype-devel fontconfig-devel libicu-devel sqlite-devel libpng-devel libjpeg-devel python-setuptools git xorg-x11-devel xorg-x11-essentials xorg-x11-fonts xorg-x11 xorg-x11-util-devel libXfont-devel libXfont1 python python-setuptools >/dev/null
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: add -q

File: PhantomJS/install.sh [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+	if command -v "phantomjs" >/dev/null; then
+		printf -- "Go : Yes" >>"$LOG_FILE"
+
+		if phantomjs version | grep -q "$PACKAGE_VERSION"; then
+			printf -- "Version : %s (Satisfied) \n" "${PACKAGE_VERSION}" | tee -a "$LOG_FILE"
+			printf -- "No update required for Go \n" | tee -a "$LOG_FILE"
+			exit 0
+		fi
+	fi
+}
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: Where is interactive shell asking for confirmation ?

File: PhantomJS/install.sh [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+		printHelp
+		exit 0
+		;;
+	d)
+		set -x
+		;;
+	v)
+		PACKAGE_VERSION="$OPTARG"
+		;;
+	o)
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: Remove o , p and add y.
Please follow the latest cadvisor !!!

File: PhantomJS/install.sh [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+
+	printf -- "Detected %s \n" "$PRETTY_NAME"
+	printf -- "Request details : PACKAGE NAME= %s , VERSION= %s \n" "$PACKAGE_NAME" "$PACKAGE_VERSION" | tee -a "$LOG_FILE"
+}
+
+# Print the usage message
+function printHelp() {
+	echo
+	echo "Usage: "
+	echo "  install.sh [-s <silent>] [-d <debug>] [-v package-version] [-o override] [-p check-prequisite]"
```
Comment (imdurgadas) [inline PhantomJS/install.sh]: This is outdated. Please follow latest scripts from my repo.

**Comment by sid226** [comment]
Added changes requested. Verified on SLES 15, RHEL 7.5
LOG FOR SLES 15 
[phantomjs-2.1.1-2018-10-12-06%3A52%3A08.log](https://github.com/imdurgadas/scripts/files/2472280/phantomjs-2.1.1-2018-10-12-06.3A52.3A08.log)

LOG FOR RHEL 7.5 
[phantomjs-2.1.1-2018-10-12-07%3A06%3A31.log](https://github.com/imdurgadas/scripts/files/2472281/phantomjs-2.1.1-2018-10-12-07.3A06.3A31.log)

**Pull request merged**